imap/ical_support.c

```diff
@@ -2957,4 +2958,20 @@ EXPORTED void icalcomponent_add_required_timezones(icalcomponent *ical)
     ptrarray_fini(&tzs);
 }
 
+EXPORTED int ical_categories_is_color(icalproperty *cprop)
+{
+    const char *categories = icalproperty_get_categories(cprop);
+    int i;
+    
+
+    if (categories[0] == '#') {
+        /* Is this #RRGGBB */
+        for (i = 1; i <= 6 && isxdigit(categories[i]); i++);
+
```

Comment:
If there is a category `#A`, then `categories[4]` does not exist, but it will be read.

Reply:
No. it will not be read.  But `#A` is not a colour.

Reply:
It correctly returns false for `#A`. But it mistakenly returns true if the category value is just `#`. I think the return condition should read `i == 7 && categories[6]...` to make sure it's a legit 6-digit hex color.

Reply:
For whatever it's worth, CSS also supports three-digit hex colours (`#RGB`) -- as a shorthand for colours who values are pairs of identical digits (i.e. `#123` is shorthand for `#112233`), which I think is different from what you would get if you treated it as a 12 bit colour and multiplied it out to 24 bits (`#102030`, right?).  Users/developers who use a lot of CSS might expect the shorthand to work here too.

But `css3_color_hex_to_name()` only supports six-digit (`#RRGGBB`) or eight-digit (`#RRGGBBAA` -- of which it drops the alpha channel) anyway, so I think this function should also require (at least) six digits to consider it a valid colour.

If we wanted to accept the three-digit CSS shorthand, I guess we'd need to extend `css3_color_hex_to_name()` too -- and be careful to do so by duplicating digits rather than just expanding 12-bit to 24-bit.  I don't think it's worth the effort, unless we see it in the wild?

Reply:
Length of the color (7) is now checked

Rik says we won't worry about #123 for now